synapse/replication/tcp/client.py

```diff
@@ -140,6 +153,10 @@ def get_streams_to_replicate(self):
             args["account_data"] = user_account_data
         elif room_account_data:
             args["account_data"] = room_account_data
+
+        # Record which streams we're in the process of subscribing to
+        self.streams_connecting = set(args.keys())
```

Comment:
beware that this method is overridden in derived classes, so this won't work right.

In any case it feels odd that a method called get_foo has side effects. should we not be managing the list of pending streams in `ClientReplicationStreamProtocol` ?

Reply:
Err, yes. Point.